### DISCUSSION TITLE
코드 리뷰 관련 아티클

### DISCUSSION THREAD

**Discussion by realswxx** [Tudy-Project/TUDY#40] [아티클자료]
### 코드리뷰 
코드의 내용을 점검(examining), `피드백`을 주는 과정

`피드백` : 오타, 버그 가능성, 개발 표준들의 의견, 좋은 코드에 대한 긍정적인 피드백

### 핵심 
단순히 코딩스타일 일관되게 유지, 예상되는 문제 조기 파악을 넘어 + 
코드에 대한 책임이 그 코드를 만든 사람 혼자에게 있지 않고 우리 모두에게 있다는 문화를 만든다.

### 장점
- 버그 조기 발견
- 개발 표준 준수(컨벤션)
- 중복 코드 방지
- 모듈의 재사용성 증대 
- 다른 사람의 잘 만들어진 코드를 보면서 배울 기회

### 코드 리뷰 시 확인 할 것
- 기능 정상 동작 여부
- 버그 조기 발견
- 가독성과 유지보수 편의성
- 개발 표준의 준수 여부
- 테스트 코드의 작성 여부
- 재사용 가능한 모둘의 중복 개발
- 배울만한 점은 없는지

### 코드 컴플릿(Code Complete)저자 스티브 맥코넬 코드리뷰의 중요성을 강조

>“소프트웨어를 유지보수하는 조직에서 코드 한 줄을 변경한다고 했을 때, 코드리뷰가 도입되기 전에는 그러한 변경의` 55%` 정도가 문제를 일으켰다. 그러나 리뷰 과정이 도입된 이후에는 그러한 변경의 `2%`정도에서만 문제가 발생했다.”

### 코드리뷰의 가장 중요한 점 한 가지

>책임자를 추궁하지 않는 문화’의 정착

[레퍼런스](https://techwell.wooritech.com/blog/2021/04/19/%EC%BD%94%EB%93%9C%EB%A6%AC%EB%B7%B0/)

**Comment by NEULiee** [comment]
잘읽었습니다!! 저는 코드에 대한 책임이 팀에게 있다는 문구가 와닿네요 ㅎㅎ
물어보면서 배울점도 많고 좋은 경험 하고 있는 것 같습니다 ㅎㅎ

**Reply by realswxx** [reply]
맞아요 저도 이 부분은 생각지 못했던 부분인지라
근데 기능 정상 작동 여부는 코드 리뷰 때 확인이 불가하지 않나요?

**Reply by NEULiee** [reply]
직접 확인하라는 것일까욥.. 코드만 보고 확인하기는 쉽지않죠
